jupyterlab/labapp.py

```diff
@@ -41,6 +41,41 @@
 if version != app_version:
     version = '%s (dev), %s (app)' % (__version__, app_version)
 
+buildFailureMsg = """Build failed.
+Troubleshooting: If the build failed due to an out-of-memory error, you
+may be able to fix it by disabling the `dev_build` and/or `minimize` options.
+
+If you are building via the `build jupyter lab` command, you can disable
+these options like so:
+
+build jupyter lab --dev-mode=False --minimize=False
+
+You can also disable these options for all JupyterLab builds by adding these
+lines to your Jupyter config file (`jupyter_notebook_config.py`):
+
+c.LabBuildApp.minimize = False
+c.LabBuildApp.dev_build = False
+
+If you haven't done so previously, you can generate a Jupyter config file by
+running:
+
+jupyter lab --generate-config
+
+Explanation:
+
+- `dev-build`: This option controls whether a `dev` or a more streamlined
+`production` build is used. This option will default to `False` (ie the
+`production` build) for most users. However, if you have any labextensions
+installed from local files, this option will instead default to `True`.
+Explicitly setting `dev-build` to `False` will ensure that the `production`
+build is used in all circumstances.
+
+- `minimize`: This option controls whether your JS bundle is minified
+during the Webpack build, which helps to improve JupyterLab's overall 
+performance. However, the minifier plugin used by Webpack is very memory
```

Comment:
"overall performance" -> "initial load times", (unless I'm misunderstanding what the minimizer does)

Reply:
You may know more than I do. Obviously minimization helps with initial load (and any subsequent lazy loading). Does it have no other benefit?